### PR TITLE
Authentication, Authorization,CRUD using Ado.net

### PR DESCRIPTION
**Pull request by MuhammadFaizanGulzar**

### CODE REVIEW

File: WEBAPI.Infrastructure/Service/UserService.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+            if (user == null || !BCrypt.Net.BCrypt.Verify(password, user.PasswordHash))
+            {
+                return null;
+            }
+
+            // Generate a strong key for token generation (256 bits)
+            //byte[] keyBytes = Generate256BitKey();
+      
+            //var key = new SymmetricSecurityKey(keyBytes);
```
Comment (taimoor0nline) [inline WEBAPI.Infrastructure/Service/UserService.cs]: Please remove the code if not needed

File: WEBAPI.Infrastructure/Service/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var key = Encoding.ASCII.GetBytes(_configuration["Jwt:Secret"]!);
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(new Claim[]
+                {
+                  new Claim("UserId", user.Id.ToString()),
+                  new Claim("UserName", user.Username.ToString()),
+                }),
+                Expires = DateTime.UtcNow.AddDays(7),
```
Comment (taimoor0nline) [inline WEBAPI.Infrastructure/Service/UserService.cs]: 7 is hard coded please get this value from some configuration settings

File: WEBAPI.Infrastructure/Service/UserService.cs
Comment (taimoor0nline) [inline WEBAPI.Infrastructure/Service/UserService.cs:1]: This class is being used for User Management and token generation, its a good practice to split it

File: WEBAPI.Infrastructure/Service/ItemService.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WEBAPI.Application.Interfaces;
```
Comment (taimoor0nline) [inline WEBAPI.Infrastructure/Service/ItemService.cs:6]: remove the unused name spaces

File: WEBAPI.Infrastructure/Service/ItemService.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        public ItemService(IOptions<AppSettings> appSettings)
+        {
+            _appSettings = appSettings.Value;
+            AppDb = new ApplicationDbContext(_appSettings.ConnectionString);
+          
+        }
+
+        public IEnumerable<Item> GetAllItems()
+        {
+            string query = "select * from Items";
```
Comment (taimoor0nline) [inline WEBAPI.Infrastructure/Service/ItemService.cs]: please consider only those columns which are required avoid using * in select

Comment (taimoor0nline) [reply WEBAPI.Infrastructure/Service/ItemService.cs]: consider having some pagination to  improve performance

File: WEBAPI.Infrastructure/DB/ApplicationDbContext.cs [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        public List<User> GetUsers()
+        {
+            List<User> _userDetails = new List<User>();
+            string connectionString = ConnectionString;
+
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+
+                string sql = "SELECT * FROM Users"; // Replace YourTableName with your actual table name
```
Comment (taimoor0nline) [inline WEBAPI.Infrastructure/DB/ApplicationDbContext.cs]: it's good to use store procedure or use abstraction

File: WEBAPI.Infrastructure/DB/ApplicationDbContext.cs [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+            }
+            catch (Exception ex)
+            {
+                // Handle exceptions appropriately
+            }
+
+            return objresutl;
+        }
+
+        public List<T> ConvertDataTableToList<T>(DataTable dt) where T : new()
```
Comment (taimoor0nline) [inline WEBAPI.Infrastructure/DB/ApplicationDbContext.cs:157]: Are not these duplicate ConvertDataTable<T>() and ConvertDataTableToList<T>

**Review by taimoor0nline — CHANGES_REQUESTED** [review]
Please review the changes

File: WEBAPI.Infrastructure/Service/UserService.cs (hunk 1 above)
Comment (MuhammadFaizanGulzar) [reply WEBAPI.Infrastructure/Service/UserService.cs]: Resolved.

File: WEBAPI.Infrastructure/Service/UserService.cs (hunk 2 above)
Comment (MuhammadFaizanGulzar) [reply WEBAPI.Infrastructure/Service/UserService.cs]: Resolved.

File: WEBAPI.Infrastructure/Service/ItemService.cs (hunk 3 above)
Comment (MuhammadFaizanGulzar) [reply WEBAPI.Infrastructure/Service/ItemService.cs:6]: Resolved.

File: WEBAPI.Infrastructure/Service/ItemService.cs (hunk 4 above)
Comment (MuhammadFaizanGulzar) [reply WEBAPI.Infrastructure/Service/ItemService.cs]: Resolved.

File: WEBAPI.Infrastructure/DB/ApplicationDbContext.cs (hunk 5 above)
Comment (MuhammadFaizanGulzar) [reply WEBAPI.Infrastructure/DB/ApplicationDbContext.cs]: Resolved.

File: WEBAPI.Infrastructure/DB/ApplicationDbContext.cs (hunk 6 above)
Comment (MuhammadFaizanGulzar) [reply WEBAPI.Infrastructure/DB/ApplicationDbContext.cs:157]: Yes they were duplicate, removed One.

File: WEBAPI.Infrastructure/Service/UserService.cs
Comment (MuhammadFaizanGulzar) [reply WEBAPI.Infrastructure/Service/UserService.cs:1]: Done, created separate service for that.

**Review by taimoor0nline — APPROVED** [review]

**Pull request closed without merging**